modules/http-api/src/main/java/org/mule/runtime/http/api/client/HttpRequestOptions.java

```diff
@@ -0,0 +1,51 @@
+/*
+ * Copyright (c) MuleSoft, Inc.  All rights reserved.  http://www.mulesoft.com
+ * The software in this package is published under the terms of the CPAL v1.0
+ * license, a copy of which has been included with this distribution in the
+ * LICENSE.txt file.
+ */
+package org.mule.runtime.http.api.client;
+
+import org.mule.api.annotation.NoImplement;
+import org.mule.runtime.http.api.client.auth.HttpAuthentication;
+import org.mule.runtime.http.api.domain.message.request.HttpRequest;
+
+/**
+ * Options for setting up an {@link HttpRequest}. Instances can only be obtained through a {@link HttpRequestOptionsBuilder}.
+ *
+ * @since 4.2
+ */
+@NoImplement
```

Comment:
isn't this to be implemented by the service implelemtation?

Reply:
If it was any additions to it would require default implementations but that can be handled by the builder which is in place to make usage easier.